src/core/halide_utils.cc

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "tc/core/halide_utils.h"
+
+#include <map>
+#include <vector>
+
+#include "tc/core/flags.h"
+#include "tc/core/tc2halide.h"
+#include "tc/core/utils/dlpack.h"
+
+namespace tc {
+
+using namespace dlutils;
+using namespace Halide;
+using namespace Halide::Internal;
+
+DLDataType fromHalideType(const Halide::Type& type) {
+  // Hmm, these are suspiciously similar...
```

Comment:
haha, tell me about it. You'll want to check that the code enum matches between Halide and DLType though.